### DISCUSSION TITLE
Request: allow user to determin LONG PRESS duration

### DISCUSSION THREAD

**Discussion by andyblac** [mongoose-os-apps/shelly-homekit#1467] [Ideas]
can we add support to control how long the button has to be held down to action as a LONG PRESS, when in detached mode.

Some familly members and guests are holding the button to long and it is being detected as a LONG PRESS, but they just wanted a SHORT PRESS, it would be good if we can allow control of this in the webUI.

@ZeevoX is this something you could take a look at ?

**Comment by rojer** [comment]
makes sense

**Comment by andyblac** [comment]
yeah maybe @timoschilling or @ZeevoX could take a look.

**Comment by timoschilling** [comment]
@rojer could you point out the place where the value is set/used?

**Comment by andyblac** [comment]
i found these

https://github.com/mongoose-os-apps/shelly-homekit/blob/eb9dd677e677b9180a9f37eecf7b523cceeed81c/src/shelly_input_pin.cpp#L31
https://github.com/mongoose-os-apps/shelly-homekit/blob/47e0c4673637c3b7b5e5d6ec8022195de0a6c912/src/shelly_input_pin.hpp#L30
https://github.com/mongoose-os-apps/shelly-homekit/blob/ccf11f9e6c20cfc6136359956dfe81a2c4f5786c/src/shelly_main.cpp#L729

**Comment by rojer** [comment]
default values are constants - set [here](https://github.com/mongoose-os-apps/shelly-homekit/blob/22a1d9a84f0083dd4edd8188459249ed8f71a0ff/src/shelly_input_pin.cpp#L30-L31)
